lib/chef/client.rb

```diff
@@ -327,12 +327,12 @@ def run
     def warn_if_eol
       require_relative "version"
 
-      # We make a release every year so take the version you're on + 2006 and you get
+      # We make a release every year so take the version you're on + 2007 and you get
       # the year it goes EOL
-      eol_year = 2006 + Gem::Version.new(Chef::VERSION).segments.first
+      eol_year = 2007 + Gem::Version.new(Chef::VERSION).segments.first
 
-      if Time.now > Time.new(eol_year, 5, 01)
-        logger.warn("This release of #{ChefUtils::Dist::Infra::PRODUCT} became end of life (EOL) on May 1st #{eol_year}. Please update to a supported release to receive new features, bug fixes, and security updates.")
+      if Time.now > Time.new(eol_year, 12, 31)
```

Comment:
The EOL dat for chef-16 is set for Nov 30th 2022.

Reply:
@johnmccrae @PrajaktaPurohit  Was there specific reason to update 2006 to 2007 ?
the `eol_year` is getting calculated incorrectly to 2023 otherwise instead of 2022